src/sentry/incidents/logic.py

```diff
@@ -300,13 +303,25 @@ def bulk_build_incident_query_params(incidents, start=None, end=None, prewindow=
 
     query_args_list = []
     for incident in incidents:
+        # TODO: When time_window is persisted, switch to using that instead of alert_rule.time_window.
+        time_window = incident.alert_rule.time_window if incident.alert_rule is not None else 1
+        time_window_delta = timedelta(minutes=time_window)
         params = {
-            "start": incident.date_started if start is None else start,
+            "start": incident.date_started - time_window_delta if start is None else start,
             "end": incident.current_end_date if end is None else end,
         }
-        if prewindow:
-            prewindow_time_range = calculate_incident_prewindow(params["start"], params["end"])
-            params["start"] = params["start"] - prewindow_time_range
+        if windowed_stats:
+            now = timezone.now()
+            params["end"] = params["start"] + timedelta(
+                minutes=time_window * (WINDOWED_STATS_DATA_POINTS / 2)
+            )
+            params["start"] = params["start"] - timedelta(
+                minutes=time_window * (WINDOWED_STATS_DATA_POINTS / 2)
+            )
+            if params["end"] > now:
```

Comment:
Should we align start/end along a boundary? like 1 minute, 10 minute, etc block? Or does that happen elsewhere?

Reply:
Am I correct in assuming the start would always be on the minute? I can definitely look into aligning the 'end' since it's just set to the current time if the incident started recently.

Reply:
No, the subscription runs every minute, but we're not guaranteed that it's exactly on the minute.